code/modules/reagents/chemistry/holder.dm

```diff
@@ -677,138 +727,325 @@
 				else
-					if(!C.required_container)
+					if(cached_my_atom.type == C.required_container)
 						matching_container = TRUE
-					if(!C.required_other)
-						matching_other = TRUE
-
-				if(required_temp == 0 || (is_cold_recipe && chem_temp <= required_temp) || (!is_cold_recipe && chem_temp >= required_temp))
-					meets_temp_requirement = TRUE
+				if (isliving(cached_my_atom) && !C.mob_react) //Makes it so certain chemical reactions don't occur in mobs
+					matching_container = FALSE
+				if(!C.required_other)
+					matching_other = TRUE
 
-				if(total_matching_reagents == total_required_reagents && total_matching_catalysts == total_required_catalysts && matching_container && matching_other && meets_temp_requirement)
-					possible_reactions  += C
+				else if(istype(cached_my_atom, /obj/item/slime_extract))
+					var/obj/item/slime_extract/M = cached_my_atom
 
-		if(possible_reactions.len)
-			var/datum/chemical_reaction/selected_reaction = possible_reactions[1]
-			//select the reaction with the most extreme temperature requirements
-			for(var/V in possible_reactions)
-				var/datum/chemical_reaction/competitor = V
-				if(selected_reaction.is_cold_recipe) //if there are no recipe conflicts, everything in possible_reactions will have this same value for is_cold_reaction. warranty void if assumption not met.
-					if(competitor.required_temp <= selected_reaction.required_temp)
-						selected_reaction = competitor
+					if(M.Uses > 0) // added a limit to slime cores -- Muskets requested this
+						matching_other = TRUE
+			else
+				if(!C.required_container)
+					matching_container = TRUE
+				if(!C.required_other)
+					matching_other = TRUE
+
+			if(required_temp == 0 || (is_cold_recipe && chem_temp <= required_temp) || (!is_cold_recipe && chem_temp >= required_temp))
+				meets_temp_requirement = TRUE
+
+			if(total_matching_reagents == total_required_reagents && total_matching_catalysts == total_required_catalysts && matching_container && matching_other && meets_temp_requirement)
+				possible_reactions  += C
+
+	update_previous_reaction_list()
+	//This is the point where we have all the possible reactions from a reagent/catalyst point of view, so we set up the reaction list
+	for(var/pr in possible_reactions)
+		var/datum/chemical_reaction/selected_reaction = pr
+		if((selected_reaction.reaction_flags & REACTION_INSTANT) || (flags & INSTANT_REACT)) //If we have instant reactions, we process them here
+			instant_react(selected_reaction)
+			.++
+			update_total()
+			continue
+		else
+			var/exists = FALSE
+			for(var/datum/equilibrium/E in reaction_list)
+				if(E.reaction.type == selected_reaction.type) //Don't add duplicates
+					exists = TRUE
+
+			//Add it if it doesn't exist in the list
+			if(!exists)
+				is_reacting = TRUE//Prevent any on_reaction() procs from infinite looping
+				var/datum/equilibrium/E = new /datum/equilibrium(selected_reaction, src) //Otherwise we add them to the processing list.
```

Comment:
```suggestion
				var/datum/equilibrium/E = new (selected_reaction, src) //Otherwise we add them to the processing list.
```
No need to duplicate the information, the variable type will be used.